nemo/collections/asr/modules/rnnt.py

```diff
@@ -162,22 +162,14 @@ def __init__(
             dropout=dropout,
         )
 
-        # Flag needed for RNNT export support
-        self._rnnt_export = False
-
     @typecheck()
     def forward(self, targets, target_length, states=None):
         # y: (B, U)
         y = rnn.label_collate(targets)
 
         # state maintenance is unnecessary during training forward call
         # to get state, use .predict() method.
-        if self._rnnt_export:
-            add_sos = False
-        else:
-            add_sos = True
-
-        g, states = self.predict(y, state=states, add_sos=add_sos)  # (B, U, D)
+        g, states = self.predict(y, state=states, add_sos=self.training)  # (B, U, D)
```

Comment:
This is incorrect - even during eval mode, we need an additional sos token in the forward. Only during export it should be one token (whatever it may be).

Reply:
Oh I got confused by the comment above. Will put that back. They really should have had it tri-state (train/eval/infer) rather than boolean.